modules/solid_mechanics/neml2/src/LAROMANCE6DInterpolation.cxx

```diff
@@ -0,0 +1,409 @@
+  _wall_grid = json_vector_to_torch("in_wall");
+  _env_grid = json_vector_to_torch("in_env");
+
+  // Read in grid axes transform enums
+  _stress_transform_enum = get_transform_enum(json_to_string("in_stress_transform_type"));
+  _temperature_transform_enum = get_transform_enum(json_to_string("in_temperature_transform_type"));
+  _plastic_strain_transform_enum =
+      get_transform_enum(json_to_string("in_plastic_strain_transform_type"));
+  _cell_transform_enum = get_transform_enum(json_to_string("in_cell_transform_type"));
+  _wall_transform_enum = get_transform_enum(json_to_string("in_wall_transform_type"));
+  _env_transform_enum = get_transform_enum(json_to_string("in_env_transform_type"));
+
+  // Read in grid axes transform values
+  _stress_transform_values = json_to_vector("in_stress_transform_values");
+  _temperature_transform_values = json_to_vector("in_temperature_transform_values");
+  _plastic_strain_transform_values = json_to_vector("in_plastic_strain_transform_values");
+  _cell_transform_values = json_to_vector("in_cell_transform_values");
+  _wall_transform_values = json_to_vector("in_wall_transform_values");
+  _env_transform_values = json_to_vector("in_env_transform_values");
+
+  // Storing values for interpolation
+  _output_rate_name = options.get<std::string>("model_file_variable_name");
+  _grid_values = json_6Dvector_to_torch(_output_rate_name);
+
+  // set up output transforms
+  if (_output_rate_name == "out_ep")
+  {
+    _output_transform_enum = get_transform_enum(json_to_string("out_strain_rate_transform_type"));
+    _output_transform_values = json_to_vector("out_strain_rate_transform_values");
+  }
+  else if (_output_rate_name == "out_cell")
+  {
+    _output_transform_enum = get_transform_enum(json_to_string("out_cell_rate_transform_type"));
+    _output_transform_values = json_to_vector("out_cell_rate_transform_values");
+  }
+  else if (_output_rate_name == "out_wall")
+  {
+    _output_transform_enum = get_transform_enum(json_to_string("out_wall_rate_transform_type"));
+    _output_transform_values = json_to_vector("out_wall_rate_transform_values");
+  }
+  else
+  {
+    throw NEMLException("This ouput variable is not implemented, model_file_variable_name: " +
+                        std::string(_output_rate_name));
+  }
+}
+
+void
+LAROMANCE6DInterpolation::request_AD()
+{
+  // only using first derivatives of out_ep, not out_cell and out_wall
+  if (_output_rate_name == "out_ep")
+  {
+    std::vector<const VariableBase *> inputs = {&_s};
+    _output_rate.request_AD(inputs);
+  }
+}
+
+void
+LAROMANCE6DInterpolation::set_value(bool out, bool dout_din, bool d2out_din2)
+{
+  neml_assert_dbg(!dout_din || !d2out_din2,
+                  "Only AD derivatives are currently supported for this model");
+  if (out)
+    _output_rate = interpolate_and_transform();
+}
+
+LAROMANCE6DInterpolation::TransformEnum
+LAROMANCE6DInterpolation::get_transform_enum(const std::string & name) const
```

Comment:
this sure is easier with `MooseEnum` and `getEnum`

Reply:
I can't use those, right?